From 1e4b75d63f2b2ccabfffbecf0bbfda42b7200dd5 Mon Sep 17 00:00:00 2001
From: Dirk Neukirchen <dirkneukirchen@web.de>
Date: Thu, 13 Nov 2014 17:49:01 +0100
Subject: [PATCH 005/310] seeks: fix build of 0.3.4

- fix MIPS16 error
{standard input}: Assembler messages:
{standard input}:288: Error: unrecognized opcode `sync'

protoc is called in Makefile.am directly
change that to some automagic/manual setting of protoc location

Signed-off-by: Dirk Neukirchen <dirkneukirchen@web.de>
---
 net/seeks/Makefile                               |  3 +
 net/seeks/patches/001-protoc_host.patch          | 77 ++++++++++++++++++++++++
 net/seeks/patches/050-better_icu_detection.patch |  7 +--
 3 files changed, 83 insertions(+), 4 deletions(-)
 create mode 100644 net/seeks/patches/001-protoc_host.patch

diff --git a/net/seeks/Makefile b/net/seeks/Makefile
index 49fd8e6..b43955e 100644
--- a/net/seeks/Makefile
+++ b/net/seeks/Makefile
@@ -18,6 +18,8 @@ PKG_SOURCE_PROTO:=git
 PKG_REV:=e8565b98cfe40fba644f587536f15f1a8665d230
 PKG_SOURCE_VERSION:=$(PKG_REV)
 
+PKG_USE_MIPS16:=0
+
 include $(INCLUDE_DIR)/package.mk
 include $(INCLUDE_DIR)/nls.mk
 
@@ -52,6 +54,7 @@ endef
 CONFIGURE_VARS:= \
 	 CXXFLAGS="-I$(STAGING_DIR)/usr/include $(ICONV_CFLAGS) $(TARGET_CFLAGS)" \
 	 LDFLAGS="-L$(STAGING_DIR)/usr/lib      $(ICONV_LDFLAGS)" \
+	 ac_cv_path_PROTOC=$(STAGING_DIR_HOST)/usr/bin/protoc \
 
 CONFIGURE_ARGS += \
 	--prefix=/usr \
diff --git a/net/seeks/patches/001-protoc_host.patch b/net/seeks/patches/001-protoc_host.patch
new file mode 100644
index 0000000..133ae9d
--- /dev/null
+++ b/net/seeks/patches/001-protoc_host.patch
@@ -0,0 +1,77 @@
+--- a/configure.in
++++ b/configure.in
+@@ -295,6 +295,11 @@ else
+ fi
+ 
+ #==========================================================================
++# Finding protoc
++#==========================================================================
++AC_PATH_PROG(PROTOC, protoc, no)
++
++#==========================================================================
+ # Finding curl
+ #==========================================================================
+ # First, check for the curl-config tool
+--- a/src/plugins/query_capture/Makefile.am
++++ b/src/plugins/query_capture/Makefile.am
+@@ -1,5 +1,7 @@
+ #automake for query_capture plugin.
+ 
++PROTOC = @PROTOC@
++
+ querycapturepluginlibdir = $(libdir)/seeks/plugins/query_capture
+ querycaptureplugindatadir = $(datadir)/seeks/plugins/query_capture
+ 
+@@ -14,7 +16,7 @@ ACLOCAL_AMFLAGS=-I m4
+ protoc_inputs=db_query_record_msg.proto
+ protoc_outputs=db_query_record_msg.pb.cc db_query_record_msg.pb.h
+ $(protoc_outputs): $(protoc_inputs)
+-	protoc -I$(srcdir) -I$(srcdir)/../../proxy/ --cpp_out=. $<
++	$(PROTOC) -I$(srcdir) -I$(srcdir)/../../proxy/ --cpp_out=. $<
+ 
+ querycapturepluginlib_LTLIBRARIES=libquerycaptureplugin.la
+ dist_libquerycaptureplugin_la_SOURCES=query_capture.cpp db_query_record.cpp \
+@@ -34,4 +36,4 @@ nobase_dist_querycaptureplugindata_DATA=
+ EXTRA_DIST = \
+ 	$(protoc_inputs)
+ 
+-SUBDIRS = . tests
+\ No newline at end of file
++SUBDIRS = . tests
+--- a/src/plugins/uri_capture/Makefile.am
++++ b/src/plugins/uri_capture/Makefile.am
+@@ -1,5 +1,7 @@
+ #automake for uri_capture plugin.
+ 
++PROTOC = @PROTOC@
++
+ uricapturepluginlibdir = $(libdir)/seeks/plugins/uri_capture
+ uricaptureplugindatadir = $(datadir)/seeks/plugins/uri_capture
+ 
+@@ -13,7 +15,7 @@ ACLOCAL_AMFLAGS=-I m4
+ protoc_inputs = db_uri_record_msg.proto
+ protoc_outputs = db_uri_record_msg.pb.cc db_uri_record_msg.pb.h
+ $(protoc_outputs): $(protoc_inputs)
+-	protoc -I$(srcdir) -I$(srcdir)/../../proxy/ --cpp_out=. $<
++	$(PROTOC) -I$(srcdir) -I$(srcdir)/../../proxy/ --cpp_out=. $<
+ 
+ uricapturepluginlib_LTLIBRARIES=liburicaptureplugin.la
+ liburicaptureplugin_la_SOURCES=uri_capture.cpp db_uri_record.cpp uri_capture.h \
+@@ -32,4 +34,4 @@ nobase_dist_uricaptureplugindata_DATA=ur
+ EXTRA_DIST = \
+ 	$(protoc_inputs)
+ 
+-SUBDIRS = . tests
+\ No newline at end of file
++SUBDIRS = . tests
+--- a/src/proxy/Makefile.am
++++ b/src/proxy/Makefile.am
+@@ -34,7 +34,7 @@ protoc_inputs=db_record_msg.proto
+ protoc_outputs=db_record_msg.pb.h db_record_msg.pb.cc
+ 
+ $(protoc_outputs): $(protoc_inputs)
+-	protoc -I$(srcdir) --cpp_out=. $<
++	$(PROTOC) -I$(srcdir) --cpp_out=. $<
+ 
+ dist_libseeksuserdb_la_SOURCES=db_record.cpp db_obj.cpp user_db.cpp
+ nodist_libseeksuserdb_la_SOURCES=$(protoc_outputs)
diff --git a/net/seeks/patches/050-better_icu_detection.patch b/net/seeks/patches/050-better_icu_detection.patch
index 7d2fa63..bc72596 100644
--- a/net/seeks/patches/050-better_icu_detection.patch
+++ b/net/seeks/patches/050-better_icu_detection.patch
@@ -1,7 +1,6 @@
-diff -ru seeks-0.3.3/configure.in seeks-0.3.3-new/configure.in
---- seeks-0.3.3/configure.in	2011-04-30 10:16:57.000000000 +0200
-+++ seeks-0.3.3-new/configure.in	2011-05-28 14:44:13.094516755 +0200
-@@ -351,17 +351,25 @@
+--- a/configure.in
++++ b/configure.in
+@@ -356,17 +356,25 @@ fi
  #==========================================================================
  # Finding icu
  #==========================================================================
-- 
2.5.0

